core/src/main/java/com/datastax/oss/driver/api/core/tracker/DistributedTraceIdGenerator.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.core.tracker;
+
+import com.datastax.oss.driver.api.core.session.Request;
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+public interface DistributedTraceIdGenerator {
+  /**
+   * Generates a unique identifier for the session request. This identifier will be added to logs.
```

Comment:
I think we should be more descriptive here, it was not obvious to me on the first read .

Something along the lines of:
- sessionRequestId: an identifier for an entire `session.execute()` call
- nodeRequestId: an identifier for the execution of a statement against a particular node. There can be multiple nodeRequestId for a given sessionRequestId (because of retries and speculative executions). Can optionally be sent to the node in the custom payload.

Both are propagated to request trackers (btw, we should rename the corresponding parameters in `RequestTracker` and adjust the descriptions).